PukingPredator/Assets/Scripts/Eating.cs

```diff
@@ -1,102 +1,131 @@
 using UnityEngine;
 
-[RequireComponent(typeof(Rigidbody))]
+[RequireComponent(typeof(Rigidbody))]
 public class Eating : InputBehaviour
 {
     /// <summary>
     /// The mass of the player before having eaten anything.
     /// </summary>
     private float baseMass;
 
-    /// <summary>
-    /// The layers that consumable objects can be on.
-    /// </summary>
-    [SerializeField]
-    private LayerMask consumableLayers;
-
-    /// <summary>
-    /// The players inventory.
-    /// </summary>
-    [SerializeField]
-    private Inventory inventory;
-
-    /// <summary>
-    /// The distance that items spawn ahead of the player when puking. (TEMP)
-    /// </summary>
-    private float pukeDistance = 0.4f;
-
-    /// <summary>
-    /// The rigidbody of the player.
-    /// </summary>
-    private Rigidbody rb;
+    /// <summary>
+    /// Default size of the character.
+    /// </summary>
+    private Vector3 baseScale;
 
     /// <summary>
-    /// The scale factor for the player's mass as they eat objects
+    /// The layers that consumable objects can be on.
     /// </summary>
-    private const float MASSFACTOR = 0.05f;
+    [SerializeField]
+    private LayerMask consumableLayers;
 
-    private Vector3 baseScale;
+    /// <summary>
+    /// The players inventory.
+    /// </summary>
+    [SerializeField]
+    private Inventory inventory;
+
+    /// <summary>
+    /// Multiplier for the mass of items in the players inventory. 0.05 means 5%
+    /// of the weight of items is added to the players mass when consumed.
+    /// </summary>
+    private const float MASS_FACTOR = 0.05f;
+
+    /// <summary>
+    /// The distance that items spawn ahead of the player when puking. (TEMP)
+    /// </summary>
+    private float pukeDistance = 2f;
+
+    /// <summary>
+    /// Force applied to object when puked. Depends on how long the puke button
+    /// was held down for.
+    /// </summary>
+    private float pukeForce
+    {
+        get {
+            float holdPercent = Mathf.Clamp(gameInput.pukeHoldDuration / MAX_PUKE_DURATION, 0, 1);
+            return Mathf.Lerp(MIN_PUKE_FORCE, MAX_PUKE_FORCE, holdPercent);
+        }
+    }
+    private const float MIN_PUKE_FORCE = 4f;
+    private const float MAX_PUKE_FORCE = 20f;
+    private const float MAX_PUKE_DURATION = 2f;

```

Comment:
is this the naming convention for constants that we want to move forward with for this project? I looked at Microsoft's documentation for C# and they say CamelCase, whereas it seems for Unity they say prefixes appended to constants r reasonable. This isn't an industry project though so I'm down to have our styles agree, just would like some consistency

Reply:
yeah, I think most files have just uppercase+underscores, whatever that convention is haha

Reply:
i went with it just because the previous one i was seeing was ALLUPPERCASE which isn't readable, so ALL_UPPERCASE was an easy swap. its not that important at this point, we already aren't following generic c# conventions with a lot of our stuff, and this is also a common convention, so unless we want to do massive changes to every other variable name that doesnt follow convention then i think were fine